### ISSUE TITLE
Windows app hanging when changing date to future date

### ISSUE THREAD

**Issue by skel35** [toggl/toggldesktop#2968]
### 💻 Environment
Platform: Windows
Toggl Version: 7.4.345

### 🐞 Actual behavior
Application hangs.

### 💯 Expected behavior
Application should handle the use case without hanging or crashing.

### 🔨 Steps to reproduce

1. Start the timer
2. Set the date to a future date (e.g. 2 days after today)
3. Press Stop (if warning about future date appears, ignore or close it)

**Comment by skel35** [comment]
When the issue #2523 was addressed the `SetTimeEntryStart` library method was fixed, but the `SetTimeEntryDate` method is still not handling the future dates. (Not saying it should be handled in the lib, might be easier to do it on a UI side, dunno yet).

**Comment by skel35** [comment]
This specific bug will not be relevant after #2974 is done, but there is a similar issue.
When setting date to a future date on a stopped time entry, the entry just gets deleted automatically during the next sync without any notification or feedback to the user.

**Comment by IndrekV** [comment]
Please create a new issue for the description in your last comment so we can look into this a bit more.

**Comment by skel35** [comment]
Created #2976.
Closing this one can be closed as it will be handled in scope of #2974.
